include/rsl/validators.hpp

```diff
@@ -0,0 +1,242 @@
+#pragma once
+
+#include <rsl/algorithm.hpp>
+
+#include <rclcpp/rclcpp.hpp>
+#include <tl_expected/expected.hpp>
+
+#include <fmt/ranges.h>
+#include <string>
+#include <vector>
+
+namespace rsl {
+
+/** @file */
+
+/**
+ * @brief TODO :(
+ */
+template <typename T>
+[[nodiscard]] auto unique(rclcpp::Parameter const& parameter) -> tl::expected<void, std::string> {
+    if (is_unique(parameter.get_value<std::vector<T>>())) return {};
+    return tl::make_unexpected(
+        fmt::format("Parameter '{}' must only contain unique values", parameter.get_name()));
+}
+
+/**
+ * @brief TODO :(
+ */
+template <typename T>
+[[nodiscard]] auto subset_of(rclcpp::Parameter const& parameter, std::vector<T> const& valid_values)
+    -> tl::expected<void, std::string> {
+    auto const& input_values = parameter.get_value<std::vector<T>>();
+    for (auto const& value : input_values)
+        if (!contains(valid_values, value))
+            return tl::make_unexpected(
+                fmt::format("Invalid entry '{}' for parameter '{}'. Not in set: {}", value,
+                            parameter.get_name(), fmt::join(valid_values, ", ")));
+    return {};
+}
+
+/**
+ * @brief TODO :(
+ */
+template <typename T, typename Fn>
+[[nodiscard]] auto size_cmp(rclcpp::Parameter const& parameter, size_t const size,
+                            std::string const& cmp_str, Fn const& cmp)
+    -> tl::expected<void, std::string> {
+    switch (parameter.get_type()) {
```

Comment:
This wasn't originally a switch statement but I realized that the previous `if/else if` solution was just a less expressive version of a switch statement so I made this change.